Update k8s ACL documentation (#5302)
hashicorp/consul: website/source/docs/platform/k8s/service-sync.html.md

The sync process must authenticate to both Kubernetes and Consul to read
and write services.

> 

Reviewer: Does it not read from the CLI arg anymore?

Author: Since this refers to the `consul-k8s` process that runs the sync, it currently does not support the CLI flag (seems to have been an implementation oversight). The env variable is not supported in the Helm chart currently either, but there's PR up for adding it, so that will be the recommended path moving. I chatted with Paul and he recommends this approach from a security standpoint.

For Kubernetes, a valid kubeconfig file must be provided with cluster

> and auth information. The sync process will look into the default locations

Reviewer 2: ```suggestion
and authentication information. The sync process will look into the default locations
```

for both in-cluster and out-of-cluster authentication. If `kubectl` works,
then the sync program should work.

> For Consul, if ACLs are configured on the cluster, a Consul [ACL token](/docs/guides/acl.html)

Reviewer 2: ```suggestion
For Consul, if ACLs are configured on the cluster, a Consul [ACL token](https://learn.hashicorp.com/consul/advanced/day-1-operations/acl-guide)
```

> will need to be provided. The process accepts this token by using the

Reviewer 2: Does the token need any specific privileges? Or can it be a management token with unrestricted privileges to start?

> `CONSUL_HTTP_TOKEN` environment variable. This token should be set as a

Reviewer 2: ```suggestion
[`CONSUL_HTTP_TOKEN`](docs/commands/index.html#consul_http_token) environment variable. This token should be set as a
```

> [Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/#creating-your-own-secrets)

Reviewer 2: As a rough example, would that process look like this:

```
kubectl create secret CONSUL_HTTP_TOKEN <your-token-here>
```
If yes, then maybe its worth noting you would only want to set it once and Kubernetes will distribute it for you to all Consul agents.

Author: So you will end up using `kubectl create secret`, but came name it however you would like. Reference to that secret should happen in the Helm chart so that the token can be attached to the sync process. A note: this is _not_ a client token, it's a token for the catalog sync pod that runs `consul-k8s`. Client tokens would be dealt with manually by connecting to the Consul cluster, so should be covered by the general ACL documentation.

Reviewer 2: Do you think its obvious that users should not use a client token? Maybe add a note here for now, then we can update the ACL docs later?
